scm/github_test.go

```diff
@@ -222,8 +222,75 @@ func TestEditRepoIssue(t *testing.T) {
 		Body:         body,
 	}
 
-	_, err = s.EditRepoIssue(ctx, issueNumber, opt)
+	_, err = s.UpdateIssue(ctx, issueNumber, opt)
 	if err != nil {
 		t.Fatal(err)
 	}
 }
+
+func TestRequestReviewers(t *testing.T) {
+	qfTestOrg := scm.GetTestOrganization(t)
+	accessToken := scm.GetAccessToken(t)
+	s, err := scm.NewSCMClient(zap.NewNop().Sugar(), "github", accessToken)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Set these when testing
+	opt := &scm.RequestReviewersOptions{
+		Organization: qfTestOrg,
+		Repository:   "repo-name",
+		Number:       0,
+		Reviewers:    []string{"reviewer-login"},
+	}
+
+	err = s.RequestReviewers(context.Background(), opt)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestCreateIssueComment(t *testing.T) {
+	qfTestOrg := scm.GetTestOrganization(t)
+	accessToken := scm.GetAccessToken(t)
+	s, err := scm.NewSCMClient(zap.NewNop().Sugar(), "github", accessToken)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// To use this test, the variables repository, body and issueNumber have to be set manually.
+	repository := "student-lab"
+	body := "Test"
+	issueNumber := 0
+	opt := &scm.IssueCommentOptions{
+		Organization: qfTestOrg,
+		Repository:   repository,
+		Body:         body,
+	}
+	_, err = s.CreateIssueComment(context.Background(), issueNumber, opt)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestUpdateIssueComment(t *testing.T) {
+	qfTestOrg := scm.GetTestOrganization(t)
+	accessToken := scm.GetAccessToken(t)
+	s, err := scm.NewSCMClient(zap.NewNop().Sugar(), "github", accessToken)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// To use this test, the variables repository, body and commentID have to be set manually.
+	repository := "student-lab"
```

Comment:
I experimented with making these tests easier to use:

```go
// Helper function that creates a new issue, and returns a cleanup function that deletes the issue.
func createAndDeleteIssue(t *testing.T, s scm.SCM, org string, repo string) (*scm.Issue, func(issueNumber int, org string, repo string)) {
	t.Helper()
	issue, err := s.CreateIssue(context.Background(), &scm.CreateIssueOptions{
		Organization: org,
		Repository:   repo,
		Title:        "Test Issue",
		Body:         "Test Body",
	})
	if err != nil {
		t.Fatal(err)
	}

	return issue, func(issueNumber int, org string, repo string) {
		accessToken := scm.GetAccessToken(t)
		s := scm.NewGithubV4SCMClient(zap.NewNop().Sugar(), accessToken)
		if err := s.DeleteIssue(context.Background(), &scm.RepositoryOptions{Owner: org, Path: repo}, issueNumber); err != nil {
			t.Fatal(err)
		}
	}
}
```

Which can be used in most of the tests similar to this:

```go
func TestCreateIssueComment(t *testing.T) {
	qfTestOrg := scm.GetTestOrganization(t)
	accessToken := scm.GetAccessToken(t)
	qfTestUser := scm.GetTestUser(t)
	s, err := scm.NewSCMClient(zap.NewNop().Sugar(), "github", accessToken)
	if err != nil {
		t.Fatal(err)
	}

	body := "Test"
	opt := &scm.IssueCommentOptions{
		Organization: qfTestOrg,
		Repository:   ag.StudentRepoName(qfTestUser),
		Body:         body,
	}

	issue, delete := createAndDeleteIssue(t, s, opt.Organization, opt.Repository)
	defer delete(issue.IssueNumber, opt.Organization, opt.Repository)

	_, err = s.CreateIssueComment(context.Background(), issue.IssueNumber, opt)
	if err != nil {
		t.Fatal(err)
	}
}
```

This will create a new issue on the `QF_TEST_ORG` for the user `QF_TEST_USER`, and delete the issue when the test returns.

Reply:
Nice one. I think you can even simplify this further by having the returned function reuse the passed in `s` if you use the `scm.GithubV4SCM` type instead of `scm.SCM`:

```go
	return issue, func(issueNumber int, org string, repo string) {
		if err := s.DeleteIssue(context.Background(), &scm.RepositoryOptions{Owner: org, Path: repo}, issueNumber); err != nil {
			t.Fatal(err)
		}
	}
```